### PR TITLE
Docs: Update merge a pull request guideline

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Update merge a pull request guideline to explain a milestone **should** be added to PRs (and (soon) will be required).

**Which issue(s) this PR fixes**:
Ref https://github.com/grafana/grafana-release/issues/60
Ref https://github.com/grafana/grafana-release/issues/55

**Special notes for your reviewer**:

### CODE REVIEW

File: contribute/merge-pull-request.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ The following checklist/summary should give you a quick overview of what to ask/
 
 Before actually merging a pull request there's a couple of things to take into consideration.
 
+### Status checks
+
+To be able to merge a pull request all of its required status checks needs to pass including a review approval.
```
Comment (achatterjee-grafana) [inline contribute/merge-pull-request.md:26]:
```suggestion
Before you can merge a pull request, it must have a review approval, and all the required status checks must pass.

```

File: contribute/merge-pull-request.md [hunk 2]
```diff
@@ -21,6 +21,10 @@ The following checklist/summary should give you a quick overview of what to ask/
...
+
 ### Format the pull request title
 
 Make sure that the pull request title is properly formatted according to `<Area>: <Summary>` and try to make the summary short and understandable for the community as a whole.
```
Comment (achatterjee-grafana) [inline contribute/merge-pull-request.md:30]:
```suggestion
The pull request title should be formatted according to `<Area>: <Summary>`. Keep the summary short and understandable for the community as a whole.
```

File: contribute/merge-pull-request.md [hunk 3]
```diff
@@ -31,9 +35,9 @@ See [formatting guidelines](create-pull-request.md#formatting-guidelines) for mo
 
 ### Assign a milestone
 
-It's recommended to add a milestone to every pull request. This makes it easier to track what changes did go into a certain release. Without this you're basically left with going through git commits which could be a lot harder.
+A milestone **should** be added to every pull request. Several things in the Grafana release process requires at least pull requests to be in a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).
 
-There's also various tooling built that in some cases requires a pull request to be assigned a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).
+In addition this makes it easier to track what changes did go into a certain release. Without this you're basically left with going through git commits which could be a lot harder.
```
Comment (achatterjee-grafana) [inline contribute/merge-pull-request.md:40]:
```suggestion
This makes it easier to track what changes go into a certain release. Without this information, release managers have to go through git commits which is not an efficient process.
```

File: contribute/merge-pull-request.md [hunk 4]
```diff
@@ -31,9 +35,9 @@ See [formatting guidelines](create-pull-request.md#formatting-guidelines) for mo
 
 ### Assign a milestone
 
-It's recommended to add a milestone to every pull request. This makes it easier to track what changes did go into a certain release. Without this you're basically left with going through git commits which could be a lot harder.
+A milestone **should** be added to every pull request. Several things in the Grafana release process requires at least pull requests to be in a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).
```
Comment (achatterjee-grafana) [inline contribute/merge-pull-request.md:38]:
```suggestion
We recommended that you add a milestone to every pull request. Several things in the Grafana release process require at least one pull request to be in a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

File: contribute/merge-pull-request.md (hunk 4 above)
Comment (marefr) [reply contribute/merge-pull-request.md:38]: "require at least one pull request to be in a milestone" is not true. Soon there will be a required check on pull requests that a milestone has been added so I want this to read that it's required.

**Review by achatterjee-grafana — APPROVED** [review]

**Pull request merged**